test/src/unit-cppapi-var-offsets.cc

```diff
@@ -354,20 +375,19 @@ void write_dense_array(
   }
 
   // TODO: remove this when sc21681 is fixed
-  serialized_writes = false;
-  if (!serialized_writes) {
-    CHECK_NOTHROW(query.submit());
-    query.finalize();
-  } else {
-    test::submit_and_finalize_serialized_query(ctx, query);
-  }
+  // Submit query
+  test::ServerQueryBuffers server_buffers_;
+  auto rc = test::submit_query_wrapper(
```

Comment:
So, this was already uncommented as part of https://github.com/TileDB-Inc/TileDB/pull/3603 , I just adapted it the same way as everything else. I don't know exactly why that works while the sparse reader corresponding case fails. I updated the comment to hint to the person that will fix sc21681 to check why that works though so that I don't lose time right now. Let me know if that works.

Reply:
I just disabled it in the end because I was getting ASAN failures, I don't know why it was enabled in the first place...